tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -243,5 +257,68 @@ private static void WriteRequestHeaders(WebRequest request, IDictionary<string,
             var proxyInstance = obj.DuckAs<ILambdaContext>();
             return proxyInstance?.ClientContext?.Custom;
         }
+
+        private class ConverterContractResolver : DefaultContractResolver
+        {
+            protected override JsonContract CreateContract(Type objectType)
+            {
+                // Create a contract with custom settings
+                var contract = base.CreateContract(objectType);
+
+                if (objectType == typeof(MemoryStream))
+                {
+                    contract.Converter = MemoryStreamConverter;
+                }
+                else if (objectType == typeof(DateTime))
+                {
+                    contract.Converter = DateTimeConverter;
+                }
+
+                return contract;
+            }
+        }
+
+        private class MemoryStreamJsonConverter : JsonConverter<MemoryStream>
+        {
+            public override void WriteJson(JsonWriter writer, MemoryStream value, JsonSerializer serializer)
+            {
+                if (value == null)
+                {
+                    writer.WriteNull();
+                    return;
+                }
+
+                try
+                {
+                    var base64String = Convert.ToBase64String(value.ToArray());
```

Comment:
Is there a way to tell if the memory stream is for Kinesis or not? Asking because this might not work for general cases. So I'm thinking about making it `KinesesMemoryStremJsonConverter` and have the `CreateContract` tell the difference? if possible...

Reply:
Mmm, the only idea that comes to mind is proxying the instance and checking if a certain field exist. I wonder how easy this would be, since at the moment of receiving the `MemoryStream`, we only receive the inner most field.

I'd think that we could handle more general use cases with different ways of handling the data.

Reply:
Yeah, let's start with this. we can refactor it later when new types with MemoryStream hit us.

Reply:
The downside with `ToArray()` is that it creates an entire copy of the array in memory, which could obviously end up being quite large. It _may_ be possible to use the underlying buffer directly, without reallocating, using something like this:

```suggestion
                    var base64String = value.TryGetBuffer(out var buffer) && buffer.Array is not null
                         ? Convert.ToBase64String(buffer.Array, buffer.Offset, buffer.Count)
                         : Convert.ToBase64String(value.ToArray());
```

You might need to add an `#if NETCOREAPP3_1_OR_GREATER` though 🤔

Reply:
The only thing is that the net461 _might_ bring in extra assemblies IIRC 🤔 I dunno, Rider was just warning me 😄